### PR TITLE
Preserve OwnerReferences when updating Ingresses with a Shared Load Balancer

### PR DESCRIPTION
**Pull request by bittermandel**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

When the IngressController reconciles a Ingress object which is configured to use a Shared LoadBalancer, each field is copied over from a DeepCopy of the original Ingress and the Status.LoadBalancer field is set to the values of the Shared LoadBalancer.

During this process, the OwnerReferences field is discarded. This causes issues when using other operators, such as Cert Manager, which relies on OwnerReferences during its reconcile process.

This fix makes sure the update process copies over the OwnerReferences field as well.

### CODE REVIEW

File: USERS.md [hunk 1]
```diff
@@ -679,3 +679,8 @@ Users (Alphabetically)
       D: Zero Hash is using Cilium as CNI for networking, security and monitoring features for Kubernetes clusters 
       U: CNI/ENI Networking, Network policies, Hubble
       Q: @eugenestarchenko
+
+    * N: Molnett
```
Comment (meyskens) [inline USERS.md]: Thanks for this, but could we move this to another PR as the review process here is different. Thanks

Comment (bittermandel) [reply USERS.md]: Got it 👍🏻

**Comment by joestringer** [comment]
Hey @bittermandel , could you squash the changes together and rebase that on top of the tip of the main branch? Cilium tries to keep the git log histor tidy with a single logical change per commit in order to help with bisecting issues in the tree. From that perspective I don't see the need for four commits in order to make a single line change. Thanks :pray:

**Comment by bittermandel** [comment]
> Hey @bittermandel , could you squash the changes together and rebase that on top of the tip of the main branch? Cilium tries to keep the git log history tidy with a single logical change per commit in order to help with bisecting issues in the tree. From that perspective I don't see the need for four commits in order to make a single line change. Thanks 🙏

Absolutely, done! I'm too used to having merge + squash 😄

**Comment by bittermandel** [comment]
I have absolutely no idea why or how Git decided to add 50+ commits and add reviewers for all those, after rebasing on main. My Git skills are limited, so I will open a new PR that is not bloated with all this other stuff. sorry.

**Comment by joestringer** [comment]
No worries. Git can be a bit painful sometimes.

For reference my usual workflow is:

```
git fetch origin
git rebase --interactive origin/main
```

Then it'll open an editor that lists all of the commits from your PR. There's ways to squash them together in that editor (instructions at the bottom), then when you save+quit, git will continue to execute the rebase.
